unison-cli/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -1933,6 +1933,13 @@ notifyUser dir = \case
   RemoteProjectBranchDoesntExist host projectAndBranch ->
     pure . P.wrap $
       prettyProjectAndBranchName projectAndBranch <> "does not exist on" <> prettyURI host
+  RemoteProjectBranchDoesntExist'Push host projectAndBranch ->
+    pure . P.wrap $
+      "Pushing failed because the target"
+        <> prettyProjectAndBranchName projectAndBranch
+        <> "does not exist on"
```

Comment:
Is there something we can say here to make it clearer in the case where I:

1. Push to `@project/branch`,
2. Delete `@project/branch`
3. `push`

Because in that case it'll say that the branch doesn't exist, but in all other cases it just creates the branch for you automatically, I think this'll be confusing.

Reply:
Yeah, I agree the wording could be improved and I like explicitly saying the old branch was deleted and suggesting pushing again. How about:

![push_000](https://github.com/unisonweb/unison/assets/1627482/459add47-9f82-4af6-b3f0-18adf007f588)